### ISSUE TITLE
have trouble at distillating with VGG networks

### ISSUE THREAD

**Issue by ArmandXiao** [GeorgeCazenavette/mtt-distillation#20]
Hi,

I encountered bloating `Synthetic-LR` and zero grant loss issue using VGG models.
- including `VGG11`, `VGG13`, `VGG16`

Similar issues are
- [Negative LR](https://github.com/GeorgeCazenavette/mtt-distillation/issues/16#issue-1338006051)
- [NaN loss](https://github.com/GeorgeCazenavette/mtt-distillation/issues/6)

But, my experiments are ok with `ConvNet` and `ResNet18` using similar scripts given below.

---

Here is the snippet of the scripts
```
SCRIPT_NAME=VGG13
MODEL=VGG13
DATASET=CIFAR10
IMAGE_PER_CLASS=1

python buffer.py --dataset=$DATASET --model=$MODEL --train_epochs=50 --num_experts=100 --buffer_path=$BUFFER_PATH --data_path=$DATA_PATH >> ./results/buffer_$SCRIPT_NAME.txt

python distill.py --dataset=$DATASET --model=$MODEL --ipc=$IMAGE_PER_CLASS --syn_steps=20 --expert_epochs=3 --max_start_epoch=20 --lr_img=1000 --lr_lr=1e-05 --lr_teacher=0.01 --buffer_path=$BUFFER_PATH --data_path=$DATA_PATH >> ./results/distill_$SCRIPT_NAME.txt
```
- `VGG11` is using `CIFAR100` and `--ipc=10`
- all models are **not** using `--zca` whitening

---

![image](https://user-images.githubusercontent.com/45375764/202590836-76a08dc3-b656-4631-99f6-f8d8d09f0778.png)
![image](https://user-images.githubusercontent.com/45375764/202590934-40acabb1-cd40-4803-ba12-f4432796e166.png)

**Either the `Synthetic-LR` goes to extremely positive or negative at very begining.**

Thank you.

**Comment by 1306935707** [comment]
I have met the same issues using VGG and ResNet Model!

**Comment by GeorgeCazenavette** [comment]
Hi!

Sorry for the delay in responding.

For VGG, I would first check to see that the expert trajectories are properly training. In my experience, VGG goes to NaN loss if the learning rate is too high.

Once you have confirmed that, just try lowering all the other learning rates during distillation time as well, and just keep lowering them until you no longer get NaN loss.

I'm also not sure how the method will work if you use a network with BatchNorm as the distilling network. We never thought of a good way to handle the running mean and variance parameters.

Hope this helps, and let me know if you have any other questions!

**Comment by ArmandXiao** [comment]
Hi,

Sorry to ask again, can you provide the training parameters of VGG experiments?

The expert trajectories are oki, reaching 80% test accuracy.

I have tried many parameter settings including decreasing both `img learning rate` and `lr learning rate` as well as changing the learning rate schedule. None of them succeeds in VGG.

Please help, thank you.

**Comment by GeorgeCazenavette** [comment]
Hey,

I don't have any parameters to share for the VGG experiments since we never ran any VGG experiments.

Maybe you could try replacing the max pooling layers with average pooling?

The huge fully-connected layers in VGG might also be an issue.

Sorry I can't be of more help.
